### PR TITLE
Hw7 t01

### PR DESCRIPTION
**Pull request by mirmozavr**
Element search in nested dict

### CODE REVIEW

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Please check that you have implemented all requirements.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
In general, I am happy with the solution. But there is some controversial moment.  You have not asked how to deal with it and implement one solution, so let's do it in a different way.  

First, we need to find a class that could be treated differently in the context of the task. And after that implement handling of both behaviours.

**Comment by mirmozavr** [comment]
I had thoughts about search in strings. Like "RED" in "Roses are RED violets are Blue". Should we do this?

**Comment by Cjkjvfnby** [comment]
> I had thoughts about search in strings. Like "RED" in "Roses are RED violets are Blue". Should we do this?

No, it violates the rules.  The idea is that string can be treated as a collection. 

```python
find_occurrences({"say": "WAAAGH!"}, "A") ==  3
```

**Comment by mirmozavr** [comment]
> So if we are looking for a single non-digit symbol 

Why do you segregate symbols into different categories?

> treat strings as iterable?

Yes, it deserves it.

File: hw7/hw7_t01.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Write a function, that takes element and finds the number of occurrences
+of this element in the tree.
+
+Tree can only contains basic structures like:
+    str, list, tuple, dict, set, int, bool
+"""
+from typing import Any, Union
+
+
+def parse_collection(parsed_collection: Union[list, set, tuple], element: Any) -> int:
```
Comment (Cjkjvfnby) [inline hw7/hw7_t01.py]: Have you missed anything?

File: hw7/hw7_t01.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            counter += parse_string(value, element)
+    return counter
+
+
+def parse_string(parsed_string: str, element: str) -> int:
+    counter = 0
+    for char in parsed_string:
+        if char == element:
+            counter += 1
+    return counter
```
Comment (Cjkjvfnby) [inline hw7/hw7_t01.py]: It could be done much simpler.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: hw7/hw7_t01.py (hunk 1 above)
Comment (mirmozavr) [reply hw7/hw7_t01.py]: Yes

File: hw7/hw7_t01.py (hunk 2 above)
Comment (mirmozavr) [reply hw7/hw7_t01.py]: And yes

File: hw3/tests/test_hw3_t02.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
-from hw3.hw3_t02 import slow_calc_500_numbers
-
+"""from hw3.hw3_t02 import slow_calc_500_numbers.
```
Comment (Cjkjvfnby) [inline hw3/tests/test_hw3_t02.py]: See pytest mark features. Probably it is named skip.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**